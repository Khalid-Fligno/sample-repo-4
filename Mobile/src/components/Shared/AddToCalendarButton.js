import React from "react";
import PropTypes from "prop-types";
import { Text, StyleSheet } from "react-native";
import colors from "../../styles/colors";
import { TouchableOpacity } from "react-native-gesture-handler";
import CalenderSvg from "../../../assets/icons/calender";

const AddToCalendarButton = (props) => {
  const { onPress } = props;

  return (
    <TouchableOpacity
      onPress={onPress}
      style={styles.addToCalendarButton}
      activeOpacity={0.8}
    >
      <CalenderSvg fill={colors.themeColor.color} />
      <Text style={styles.addToCalendarButtonText}>Add to calendar</Text>
    </TouchableOpacity>
  );
};

export default AddToCalendarButton;

AddToCalendarButton.propTypes = {
  onPress: PropTypes.func.isRequired,
};

const styles = StyleSheet.create({
  addToCalendarButton: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "flex-start",
    padding: 3,
    borderWidth: 2,
    backgroundColor: colors.themeColor.themeBackgroundColor,
    borderColor: colors.grey.standard,
    borderRadius: 50,
    shadowColor: colors.charcoal.light,
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.2,
    shadowRadius: 2,
    padding: 5,
    paddingLeft: 15,
    paddingRight: 10,
  },
  addToCalendarButtonText: {
    letterSpacing: 0.5,
    fontSize: 14,
    color: colors.themeColor.color,
    marginLeft: 8,
    marginRight: 5,
    fontWeight: "600",
  },
});
