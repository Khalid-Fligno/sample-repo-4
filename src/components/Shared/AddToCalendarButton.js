import React from 'react';
import PropTypes from 'prop-types';
import { Text, TouchableOpacity, StyleSheet } from 'react-native';
import Icon from '../../components/Shared/Icon';
import fonts from '../../styles/fonts';
import colors from '../../styles/colors';

export default class AddToCalendarButton extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
    };
  }
  render() {
    const { onPress } = this.props;
    return (
      <TouchableOpacity
        onPress={onPress}
        style={styles.addToCalendarButton}
      >
        <Icon
          name="add-to-calendar"
          size={25}
          color={colors.charcoal.standard}
        />
        <Text style={styles.addToCalendarButtonText}>
          Add to calendar
        </Text>
      </TouchableOpacity>
    );
  }
}

AddToCalendarButton.propTypes = {
  onPress: PropTypes.func.isRequired,
};

const styles = StyleSheet.create({
  addToCalendarButton: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
    backgroundColor: colors.white,
    padding: 3,
    borderWidth: 2,
    borderColor: colors.charcoal.light,
    borderRadius: 4,
    shadowColor: colors.charcoal.light,
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.2,
    shadowRadius: 2,
  },
  addToCalendarButtonText: {
    fontFamily: fonts.standard,
    fontSize: 13,
    color: colors.charcoal.light,
    marginTop: 4,
    marginLeft: 5,
    marginRight: 5,
  },
});
