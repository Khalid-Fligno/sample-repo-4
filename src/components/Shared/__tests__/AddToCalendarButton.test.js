import React from 'react';
import renderer from 'react-test-renderer';
import AddToCalendarButton from '../AddToCalendarButton';

describe('Add To Calendar Button', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<AddToCalendarButton
      onPress={jest.fn()}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
