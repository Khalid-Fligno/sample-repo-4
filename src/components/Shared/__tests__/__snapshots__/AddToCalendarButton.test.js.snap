// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add To Calendar Button renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(255, 255, 255)",
      "borderColor": "rgb(65, 64, 66)",
      "borderRadius": 4,
      "borderWidth": 2,
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "opacity": 1,
      "padding": 3,
      "shadowColor": "rgb(65, 64, 66)",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 2,
      "width": 140,
    }
  }
>
  <Text />
  <Text
    style={
      Object {
        "color": "rgb(65, 64, 66)",
        "fontFamily": "GothamBook",
        "fontSize": 12,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 4,
      }
    }
  >
    Add to calendar
  </Text>
</View>
`;
